### ISSUE TITLE
latex duplicate equations

### ISSUE THREAD

**Issue by duckunique** [doxygen/doxygen#10145]
Using doxygen v1.9.6 on Centos7, a library that has around 650 small latex equations creates png files that are all associated with the very first function listed in the function repository. (The thousands of png files are generated and one-up numbered -- their content is all just the same as for the very first function.) Doing the same with doxygen v1.9.4 works perfectly. Both versions generate an identical function repository file.

(Apologies if this is an incomplete description, or if I failed to see something about this mentioned in the v1.9.7 release notes. I can't provide the source example.)

Thanks!

**Comment by albert-github** [comment]
The description is indeed a bit incomplete. You write about "latex duplicate equations" but which output format are you using. also a small example is necessary as with the current description it is impossible to recreate your problem and even though you wrote "I can't provide the source example.";
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by duckunique** [comment]
Thanks -- C code, HTML output. I'll see if I can generate an example.

**Comment by albert-github** [comment]
- note that for doxygen formulas like `a^2=b^2+c^2` and `a^2 = b^2 + c^2` are different although in the LaTeX sense they are identical
- doxygen generates depending of the setting of `HTML_COLORSTYLE` images for `LIGHT`and or `DARK` mode and for the settings `TOGGLE`,  `AUTO_DARK` and `AUTO_LIGHT`(the default) both images
- also have a look at the MathJax possibilities (`USE_MATHJAX=YES`)

**Comment by duckunique** [comment]
Thanks for your help on this. An attempt to generate a simple example with several hundred LaTeX equations failed to fail. It must be something more subtle. This week, if I have time, I'll try to delete some source files from my actual case and see if I can narrow down the offending equation(s)/file(s).

**Comment by duckunique** [comment]
Hi. I lucked into discovering: the issue is using LaTeX equations when NUM_PROC_THREADS  is not equal to exactly 1.  For example, if you set  NUM_PROC_THREADS = 0 and run doxygen on the following:
[dupbug.txt](https://github.com/doxygen/doxygen/files/11877222/dupbug.txt)
you will see that only the first of the two distinct equations are generated (twice).

Please let me know if questions. Thanks again.

**Comment by albert-github** [comment]
Thanks for the investigation.
With the example and the current master version (1.9.8 (396688347198e076597050259f4835853bdc3b44)) I was not able to reproduce it but I can imagine that there is a small threading problem regarding formulas and that an extra lock is required,

**Comment by duckunique** [comment]
Glad to help. And thanks for maintaining such a valuable development tool.

**Comment by albert-github** [comment]
- Can you try your problem also with the current master version of doxygen?
- Can you share the resulting directories (in a compressed tar file) so we can have a more detailed look.

**Comment by duckunique** [comment]
Hi. Compiled latest doxygen on Fedora 38. Was able to reproduce bug by setting NUM_PROC_THREADS = 0. Attached compressed tar with README, Doxyfile, cpp, and html results with both settings of NUM_PROC_THREADS (0 in html/ and 1 in html.thread_1/).
[demo_dupbug.tar.gz](https://github.com/doxygen/doxygen/files/11924273/demo_dupbug.tar.gz)

**Comment by albert-github** [comment]
Thanks for the valuable investigations. 
I now see that the problem is not in the number of the png files (as I thought sofar) but in the content of the png files.
I now see the problem as well and can reproduce it.
It looks like it slipped in in the 1.9.6 version, more specifically (probably ) in:
```
Commit: a833e72a33b643690a4b64531f57646abeab76e4 [a833e72]
Date: Thursday, December 22, 2022 2:32:56 PM

NUM_PROC_THREAD!=1 will now process formulas on multiple threads

Also fixed a couple of other threading issues.
```
It looks like the splitting of the generated dvi file is not done properly as the file html/_formulas.dvi is still OK but the subsequent results are not.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10158

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by duckunique** [comment]
Thanks for pushing out a solution so quickly! The new master works beautifully.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by duckunique** [comment]
Built v1.9.8 from source on Fedora 38 and did simple test and all is good, thanks!
